libretro-common/net/net_compat.c

```diff
@@ -204,7 +210,8 @@ int getaddrinfo_retro(const char *node, const char *service,
       if (!host || !host->h_addr_list[0])
          goto error;
 
-      in_addr->sin_addr.s_addr = inet_addr(host->h_addr_list[0]);
+      in_addr->sin_family = host->h_addrtype;
+      memcpy(&in_addr->sin_addr, host->h_addr, host->h_length);
```

Comment:
In theory it should be, I would recommend testing

Reply:
Inviting @GregorR in here to ask his opinion on whether or not this can do harm for all other platforms.

Reply:
... does this work on PS3? It doesn't make sense. sin_addr is just a uint32_t, whereas h_addr is a string. If this works on PS3's network stack, then PS3's gethostbyname does not implement gethostbyname.

Reply:
@clienthax I think you told me earlier that despite these changes it still doesn't work, right? Please elaborate.

Reply:
I'm not familiar with the network stack in retroarch so you would need to investigate it further, but with this change it appears that the right IP and port is passed to the network library, this can be seen via rpcs3

Reply:
Let me be abundantly clear:

           struct hostent {
               char  *h_name;            /* official name of host */
               char **h_aliases;         /* alias list */
               int    h_addrtype;        /* host address type */
               int    h_length;          /* length of address */
               char **h_addr_list;       /* list of addresses */
           }
           #define h_addr h_addr_list[0] /* for backward compatibility */

           typedef uint32_t in_addr_t;

           struct in_addr {
               in_addr_t s_addr;
           };

This code is memcpying an array of pointers over a uint32_t. h_length isn't even the size of h_addr, it's the number of (char*-sized) entries. If this accidentally works, then either gethostbyname on PS3 is insane, or something else is using getaddrinfo_retro in an insane way.

Reply:
Hello I have undone my previous post because it was all wrong please hold :)

Reply:
My apologies, I was confusing gethostbyname with a different function @_@. This is correct. Ideally it should check that it's IPv4, as it'll overrun if it's IPv6 (which it shouldn't be)

Reply:
I will just add a comment then warning anybody that in case IPv6 ever gets supported for 'legacy network' targets (very unlikely), that this will need changing.